Simplenote/Classes/SPNoteListViewController.m

```diff
@@ -511,28 +505,42 @@ - (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)
-}
-
-- (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath {
+    Note *note = [self.fetchedResultsController objectAtIndexPath:indexPath];
     
-    if (editingStyle == UITableViewCellEditingStyleDelete) {
+    UITableViewRowAction *delete = [UITableViewRowAction rowActionWithStyle:UITableViewRowActionStyleDefault title:NSLocalizedString(@"Trash-verb", @"Trash (verb) - the action of deleting a note") handler:^(UITableViewRowAction *action, NSIndexPath *indexPath)
+                                    {
+                                        // Delete something here
+                                        [SPTracker trackListNoteDeleted];
+                                        [[SPObjectManager sharedManager] trashNote:note];
+                                        [[CSSearchableIndex defaultSearchableIndex] deleteSearchableNote:note];
+                                    }];
+    delete.backgroundColor = [UIColor redColor];
+    
+    UITableViewRowAction *restore = [UITableViewRowAction rowActionWithStyle:UITableViewRowActionStyleDefault title:NSLocalizedString(@"Restore", @"Restore a note from the trash, markking it as undeleted")  handler:^(UITableViewRowAction *action, NSIndexPath *indexPath)
+                                    {
+                                        [[SPObjectManager sharedManager] restoreNote:note];
+                                        [[CSSearchableIndex defaultSearchableIndex] indexSearchableNote:note];
+                                    }];
+    restore.backgroundColor = [UIColor blueColor];
+   
+    UITableViewRowAction *permanentDelete = [UITableViewRowAction rowActionWithStyle:UITableViewRowActionStyleDefault title:NSLocalizedString(@"Delete", @"Trash (verb) - the action of deleting a note") handler:^(UITableViewRowAction *action, NSIndexPath *indexPath)
+                                    {
+                                        // Delete something here
+                                        [SPTracker trackListNoteDeleted];
+                                        [[SPObjectManager sharedManager] permenentlyDeleteNote:note];
+                                        
+                                    }];
+    permanentDelete.backgroundColor = [UIColor redColor];
     
-        // delete note at index path
-        Note *note = [self.fetchedResultsController objectAtIndexPath:indexPath];
+
+    if (tagFilterType == SPTagFilterTypeDeleted) {
         
-        if (note.deleted) {
-            [[SPObjectManager sharedManager] restoreNote:note];
-            [[CSSearchableIndex defaultSearchableIndex] indexSearchableNote:note];
-        } else {
-            [SPTracker trackListNoteDeleted];
-            [[SPObjectManager sharedManager] trashNote:note];
-            [[CSSearchableIndex defaultSearchableIndex] deleteSearchableNote:note];
-        }
+        return  @[restore,permanentDelete];
```

Comment:
Done :) the actual order was Restore-Delete but @folletto suggested we should change the order to Delete-Restore so i changed that but now the order is Restore-Delete